CMakeLists.txt

```diff
@@ -0,0 +1,218 @@
+  set(CHPL_LLVM_GCC_PREFIX $ENV{CHPL_LLVM_GCC_PREFIX} CACHE STRING "Override CHPL_LLVM_GCC_PREFIX flag" FORCE)
+else()
+  set(ENV{CHPL_LLVM_GCC_PREFIX} ${CHPL_LLVM_GCC_PREFIX})
+  message(STATUS "CHPL_LLVM_GCC_PREFIX updated to ${CHPL_LLVM_GCC_PREFIX}")
+endif()
+
+
+# If any -DCHPL_ variables came to us from the command line, update them in the
+# environment so that they can be used by printchplenv
+if (DEFINED ${CHPL_HOME})
+  set(ENV{CHPL_HOME} ${CHPL_HOME})
+  message(NOTICE "CHPL_HOME updated to ${CHPL_HOME}")
+endif()
+if (DEFINED ${CHPL_HOST_PLATFORM})
+  set(ENV{CHPL_HOST_PLATFORM} ${CHPL_HOST_PLATFORM})
+  message(STATUS "CHPL_HOST_PLATFORM updated to ${CHPL_HOST_PLATFORM}")
+endif()
+if (DEFINED ${CHPL_HOST_COMPILER})
+  set(ENV{CHPL_HOST_COMPILER} ${CHPL_HOST_COMPILER})
+  message(STATUS "CHPL_HOST_COMPILER updated to ${CHPL_HOST_COMPILER}")
+endif()
+if (DEFINED ${CHPL_HOST_CC})
+  set(ENV{CHPL_HOST_CC} ${CHPL_HOST_CC})
+  message(STATUS "CHPL_HOST_CC updated to ${CHPL_HOST_CC}")
+endif()
+if (DEFINED ${CHPL_HOST_CXX})
+  set(ENV{CHPL_HOST_CXX} ${CHPL_HOST_CXX})
+  message(STATUS "CHPL_HOST_CXX updated to ${CHPL_HOST_CXX}")
+endif()
+if (DEFINED ${CHPL_HOST_ARCH})
+  set(ENV{CHPL_HOST_ARCH} ${CHPL_HOST_ARCH})
+  message(STATUS "CHPL_HOST_ARCH updated to ${CHPL_HOST_ARCH}")
+endif()
+if (DEFINED ${CHPL_HOST_MEM})
+  set(ENV{CHPL_HOST_MEM} ${CHPL_HOST_MEM})
+  message(STATUS "CHPL_HOST_MEM updated to ${CHPL_HOST_MEM}")
+endif()
+if (DEFINED ${CHPL_LLVM})
+  set(ENV{CHPL_LLVM} ${CHPL_LLVM})
+  message(STATUS "CHPL_LLVM updated to ${CHPL_LLVM}")
+endif()
+if (DEFINED ${CHPL_LLVM_SUPPORT})
+  set(ENV{CHPL_LLVM_SUPPORT} ${CHPL_LLVM_SUPPORT})
+  message(STATUS "CHPL_LLVM_SUPPORT updated to ${CHPL_LLVM_SUPPORT}")
+endif()
+if (DEFINED ${CHPL_LLVM_CONFIG})
+  set(ENV{CHPL_LLVM_CONFIG} ${CHPL_LLVM_CONFIG})
+  message(STATUS "CHPL_LLVM_CONFIG updated to ${CHPL_LLVM_CONFIG}")
+endif()
+if (DEFINED ${CHPL_SANITIZE})
+  set(ENV{CHPL_SANITIZE} ${CHPL_SANITIZE})
+  message(STATUS "CHPL_SANITIZE updated to ${CHPL_SANITIZE}")
+endif()
+
+# Makefile arguments we still want to support
+# TODO: SEE IF CMAKE COVERS THESE VARIABLES
+# TODO: Wire these up to do something
+set(CHPL_MAKE_ASSERTS OFF CACHE BOOL "ASSERTS flag from make")
+set(CHPL_MAKE_WARNINGS OFF CACHE BOOL "WARNINGS flag from make")
+set(CHPL_MAKE_OPTIMIZE OFF CACHE BOOL "OPTIMIZE flag from make")
+set(CHPL_MAKE_DEBUG OFF CACHE BOOL "DEBUG flag from make")
+set(CHPL_MAKE_PROFILE OFF CACHE BOOL "PROFILE flag from make")
+
+
+
+set(PRINTCHPLENV_COMMAND printchplenv --compiler --internal --cmake)
+# message(STATUS "running ${PRINTCHPLENV_COMMAND} from ${CMAKE_CURRENT_SOURCE_DIR}/util")
+
+# Pass the CMake detected C/CXX compilers to printchplenv
+execute_process(COMMAND ${CMAKE_COMMAND} -E env
+                  CHPL_HOST_CC=${CMAKE_C_COMPILER} CHPL_HOST_CXX=${CMAKE_CXX_COMPILER} ${CMAKE_CURRENT_SOURCE_DIR}/util/${PRINTCHPLENV_COMMAND}
```

Comment:
Interesting. It seems that either we should support the standard cmake way of overriding these & make e.g. `cmake -DCHPL_HOST_CC=bla` be an error; or we should tell cmake to use `CHPL_HOST_CC` etc, if set, when detecting the compiler. Could you add a TODO comment about this?

Reply:
OK. In that case, later on, when we have Makefiles that call the new cmake stuff, it should set `DCMAKE_C_COMPILER` and `DCMAKE_CXX_COMPILER` based upon `CHPL_HOST_CC` / `CHPL_HOST_CXX` environment / chplconfig variables, if they are set.